### DISCUSSION TITLE
Documentation. Add mobile support or probably switch to MkDocs?

### DISCUSSION THREAD

**Discussion by donbeave** [micronaut-projects/micronaut-core#3851] [Ideas] [answered]
The current documentation https://micronaut.io/documentation.html is difficult to read using mobile devices, this is how it looks right now:

![telegram-cloud-photo-size-5-6143470836393290314-y](https://user-images.githubusercontent.com/139017/89208689-6c5aa000-d5ef-11ea-8f8e-60403276f5bb.jpg)

![telegram-cloud-photo-size-5-6143470836393290315-y](https://user-images.githubusercontent.com/139017/89208699-7086bd80-d5ef-11ea-900e-ea764006dfee.jpg)

Not very convenient for reading. I think I can help you guys with making it responsive if you need it. But personally I will suggest thinking to switch to use the MkDocs engine. Especially [Material Design theme](https://github.com/squidfunk/mkdocs-material) looks pretty good, here are some examples:

1) Kubernetes NGINX Ingress documentation: https://kubernetes.github.io/ingress-nginx/
2) Testcontainers documentation: https://www.testcontainers.org
3) Traefik: https://docs.traefik.io (good example with custom header)
4) AWS Security Workshops: https://awssecworkshops.com

Here is a list of benefits:

* it uses markdown
* responsive design out-of-box
* searchable (but static, not need servers)
* easy customize
* developed by the community, always have some nice plugins

If you want to do such changes, I can help with tune colors to make it "Micronauted".

**Comment by graemerocher** [comment, marked as answer]
Improving on mobile is a good goal, but `it uses markdown` is not an advantage since asciidoc is way superior plus all of documentation is multi-language, multi-build tool and all snippets are actual tests run against real code. 

I am sure that MkDocs is cool and all but switching to it just for mobile support is a non starter.

**Comment by donbeave** [comment]
@graemerocher Ok, got it, agree, I'll check how to use the similar theme as `mkdocs-material`, but without changing the engine, just use it with `asciidoc` snippets.

**Comment by sdelamo** [comment]
Please, note you can collapse the table of contents left sidebar by clicking on the [+] sign on the top right corner. 

<img width="375" alt="Documentation  Add mobile support or probably switch to MkDocs · Discussion #3851 · micronaut-projectsmicronaut-core" src="https://user-images.githubusercontent.com/864788/93732849-72bece80-fbd3-11ea-9840-c72d704a989a.png">

I am not sure we want to change themes yet. If you want to improve the design of the mobile or the general css. I will suggest a PR to improve the CSS. 

https://github.com/micronaut-projects/micronaut-docs/tree/master/src/main/docs/resources/css

**Reply by donbeave** [reply]
@sdelamo Here it is https://github.com/micronaut-projects/micronaut-docs/pull/81. I have tried to improve it for mobile usage. As a frequent mobile user who always uses phones for reading documentation the proposed format from my point of view is much more comfortable for reading the documentation on small devices.
